sycl/include/CL/sycl/detail/pi_api_id.hpp

```diff
@@ -0,0 +1,60 @@
+//==---------- pi_api_id.hpp - PI API function IDs -------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+/// \file pi_api_id.hpp
+/// This file contains mapping between PI API functions and their IDs. A hash
+/// function is used to generate IDs. The reason for usage of a hash function
+/// instead of raw values of PiApiKind is ABI stability. New functions can be
+/// added to PiApiKind, and there's no reliable way for external XPTI users to
+/// know when to update the values. Hashes are calculated from API function
```

Comment:
I don't quite understand the motivation here: the new PI API would be added to the end of PiApiKind, so existing values are not ever affected. XPTI users should do something fake for the piApiKind that it doesn't know about. And I don't see how using piApiID instead of piApuKind would be better in telling XPTI users "when to update".

Also, what makes the hash return unique values for all PI API function names?

Reply:
If we can guarantee, that the new values will be added to the end of the enum, I'll happily get rid of `PiApiID`.

> Also, what makes the hash return unique values for all PI API function names?

Nothing. Math tells us that the probability of hash collision is `1/m`. The uniqueness can be checked in compile time by specializing some template.

Anyway, I'm going to remove this file.

Reply:
Sounds good. Please add a comment to `PiApiKind` that the new values are to be added to the end since this is an external interface. I believe we should actually treat any external enum as such and not change the order of values.